src/oal_agent/services/storage.py

```
@@ -105,3 +108,15 @@ async def load(self, key: str):
             data = self._decrypt(data)
 
         return data
+
+    async def check_health(self) -> bool:
+        """Checks the health of the storage service."""
+        logger.debug("Checking storage service health...")
+        # Check if the storage path exists and is writable
+        if not await asyncio.to_thread(self.storage_path.is_dir):
+            logger.error("Storage path does not exist or is not a directory: %s", self.storage_path)
+            return False
+        if not await asyncio.to_thread(os.access, self.storage_path, os.W_OK):
```

Ruff (0.14.7)

Undefined name `os`

(F821)